### ISSUE TITLE
Training and Evaluating an SVM Classifier on the Digits Dataset

### ISSUE THREAD

**Issue by Pankaj-Str** [Pankaj-Str/MachineLearning#1]
**Objective**: In this exercise, you will train a Support Vector Machine (SVM) classifier using the scikit-learn digits dataset and evaluate its performance. You will explore different kernels, tune hyperparameters, and aim to achieve the highest possible accuracy.

**Dataset**: The digits dataset from scikit-learn (`sklearn.datasets.load_digits`) consists of 8x8 pixel images of handwritten digits (0-9) with 1797 samples.

**Tasks**:

1. **Load and Prepare the Data**:
   - Load the digits dataset using `from sklearn.datasets import load_digits`.
   - Split the dataset into training and testing sets, using 80% of the samples for training and 20% for testing. Ensure reproducibility by setting a random seed (e.g., `random_state=42`).

2. **Train and Evaluate SVM with Different Kernels**:
   - Train an SVM classifier using the following kernels:
     - **RBF kernel** (default kernel in `sklearn.svm.SVC`).
     - **Linear kernel**.
   - For each kernel, compute and report the accuracy score on the test set using `accuracy_score` from `sklearn.metrics`.

3. **Hyperparameter Tuning**:
   - Perform hyperparameter tuning on the SVM classifier to maximize accuracy. Focus on the following parameters:
     - **Regularization parameter (`C`)**: Controls the trade-off between achieving a low error on the training data and minimizing the margin. Try values such as `[0.1, 1, 10, 100]`.
     - **Gamma parameter (`gamma`)** (for RBF kernel only): Defines how far the influence of a single training example reaches. Try values such as `[0.001, 0.01, 0.1, 1]`.
   - Use grid search (`GridSearchCV` from `sklearn.model_selection`) to systematically evaluate combinations of these parameters.
   - Report the best hyperparameters and the corresponding highest accuracy score on the test set.

## Example Code Structure

```python 

from sklearn.datasets import load_digits
from sklearn.model_selection import train_test_split, GridSearchCV
from sklearn.svm import SVC
from sklearn.metrics import accuracy_score

# Load dataset
digits = load_digits()
X, y = digits.data, digits.target

# Split data
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Train and evaluate SVM with RBF and Linear kernels
# ... (your code here)

# Perform grid search for hyperparameter tuning
# ... (your code here)

# Report results
# ... (your code here)

````
